src/app/containers/SocialEmbed/index.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useContext } from 'react';
+import path from 'ramda/src/path';
+import styled from 'styled-components';
+import {
+  AmpSocialEmbed,
+  CanonicalSocialEmbed,
+} from '@bbc/psammead-social-embed';
+import { GEL_SPACING_TRPL } from '@bbc/gel-foundations/spacings';
+import { RequestContext } from '#contexts/RequestContext';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { GridItemConstrainedMedium } from '#lib/styledGrid';
+import useToggle from '#hooks/useToggle';
+import socialEmbedBlockPropTypes from '#models/propTypes/socialEmbed';
+import createTranslations from './translations';
+
+/**
+ * MAX_WIDTH ensures all provider embeds take up the same width.
+ * NB Tweets max-out at 500px, which is represented as 31.25rem.
+ */
+const MAX_WIDTH = '31.25rem';
+
+const Wrapper = styled.div`
+  margin-right: auto;
+  margin-left: auto;
+  margin-bottom: ${GEL_SPACING_TRPL};
+  max-width: ${MAX_WIDTH};
+
+  .twitter-tweet,
+  .instagram-media {
+    margin-top: 0 !important;
+    margin-bottom: 0 !important;
+  }
+  .instagram-media {
+    min-width: auto !important;
+  }
+`;
+
+const SocialEmbedContainer = ({ blocks }) => {
+  const { isAmp } = useContext(RequestContext);
+  const { service, translations } = useContext(ServiceContext);
+  const { enabled } = useToggle('socialEmbed');
+
+  if (!blocks || !enabled) return null;
+
+  const { type: provider, indexOfType, model } = blocks[0];
+  const { id, href } = model;
+  const index = indexOfType + 1;
+
+  const oEmbed = path(['embed', 'oembed'], model);
```

Comment:
This section makes the (perhaps safe, I don't know) assumption that all necessary properties will be always present. What is the expected behaviour if we receive a block that contains incomplete/invalid data?

Reply:
This is safe thanks to the prop-type validations we make in `models/propTypes/socialEmbed` (which you've just reminded me I need to add `indexOfType` to 👍) and the use of Ramda `path`.

Reply:
@jamesdonoh I've added a further protection, which ensures we won't render if, for some reason, `model.id` and `model.href` aren't present in the Ares response.